com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapterContext.java

```diff
@@ -175,6 +178,11 @@ public void setVariableFormatter(IVariableFormatter variableFormatter) {
         return sourceMappingCache;
     }
 
+    @Override
+    public Map<Long, StackFrame[]> getThreadStackFrameMap() {
+        return threadStackFrameMap;
+    }
+
```

Comment:
Refactor `IStackFrameProvider` -> `StackFrameCache`. And let `DebugAdapterContext` to expose `getStackFrameCache()`.